### PR TITLE
Fix experiemental CI trigger condition + Fix RL CI editable install permission error

### PR DESCRIPTION
**Pull request by wwwjn**
Fix 3 things related to experimental CI:
1. In RL integration test: editable install `torchtitan` caused permission issue
2. When first create a RL PR (with `ciflow/rl` tag), the first commit doesn't trigger the CI run, only triggered a `set-matrix`
3. `is-experimental` flag in `set-matrix` will trigger a empty ROCm matrix but runs nothing.

After fix, the CI trigger will be: 
- experimental + labeled -> CUDA only
- experimental + main push -> CUDA only
- experimental + schedule -> CUDA only
- non-experimental ROCm label/tag behavior preserved.

Other changes need to be fixed when RL CI is down:
  - Updates RL integration tests to run via python -m torchtitan.experiments.rl.train.
  - Uses varlen RL configs in CI, disables Qwen3 thinking, sets group_size=2, raises seq_len to 1024, and caps generation with generator.sampling.max_tokens=256 so rollout rows are not all filtered as overlength.
  - Splits H100 bitwise parity into separate Varlen and Flex torchrun invocations so each gets a fresh process group / vLLM lifecycle. 
  - Fixes teardown hangs by not destroying Monarch-owned distributed state from actor close() methods: VLLMGenerator.close() only shuts down renderer-local resources, and PolicyTrainer.close() leaves process group teardown to ProcMesh.stop(). cc @felipemello1 @pzhan9 
  - Routes trainer checkpoint/debug dump paths through the CI output directory to avoid permission/path issues.
  - Renames/updates RL docs and helper imports to the varlen config names.

### CODE REVIEW

File: .github/workflows/integration_test_4gpu_rl.yaml [hunk 1]
```diff
@@ -28,14 +28,14 @@ permissions:
...
   set-matrix:
+    if: github.event_name != 'pull_request' || contains(github.event.pull_request.labels.*.name, 'ciflow/rl')
     uses: ./.github/workflows/set-matrix.yaml
     with:
       is-experimental: true
 
   # Step 2: Use the dynamic matrix in the build-test job
   build-test:
     needs: set-matrix
-    if: github.event_name != 'pull_request' || contains(github.event.pull_request.labels.*.name, 'ciflow/rl')
```
Comment (wwwjn) [inline .github/workflows/integration_test_4gpu_rl.yaml:38]: This line is moved to first step `set-matrix`. This bug is only in RL integration test yaml file, didn't find the same issue in other yaml files

File: torchtitan/experiments/rl/README.md [hunk 2]
```diff
@@ -58,9 +58,9 @@ uv pip install torch vllm torchcomms  --pre \
 **NOTE:** The pre-built vLLM wheels are only compatible with CUDA 13.0, though they should work with most older CUDA versions. Alternatively, you can install the corresponding vLLM pre-built wheels directly from https://download.pytorch.org/whl/nightly/cu130, for example: `uv pip install vllm-1.0.0.dev20260219+cu130-<suffix>.whl`. Ensure the build version number (e.g., `dev20260219`) matches your PyTorch nightly installation.
 
 
-5. Install TorchTitan in editable mode:
+5. Make the TorchTitan checkout importable:
 ```bash
-uv pip install -e .
+export PYTHONPATH="$PWD:${PYTHONPATH:-}"
```
Comment (felipemello1) [inline torchtitan/experiments/rl/README.md]: this looks like an odd instruction to be in the readme. Are you confident about it?

Comment (wwwjn) [reply torchtitan/experiments/rl/README.md]: I updated the CI and want to keep README aligned. But the "$PWD" need to point to "torchtitan repository root" to be correct.. This part I'm not very confident if we have general way to indicate users in README.. Do you think we should keep `uv pip install -e .` in README?

File: torchtitan/experiments/rl/README.md [hunk 3]
```diff
@@ -58,9 +58,9 @@ uv pip install torch vllm torchcomms  --pre \
 **NOTE:** The pre-built vLLM wheels are only compatible with CUDA 13.0, though they should work with most older CUDA versions. Alternatively, you can install the corresponding vLLM pre-built wheels directly from https://download.pytorch.org/whl/nightly/cu130, for example: `uv pip install vllm-1.0.0.dev20260219+cu130-<suffix>.whl`. Ensure the build version number (e.g., `dev20260219`) matches your PyTorch nightly installation.
 
 
-5. Install TorchTitan in editable mode:
+5. From the TorchTitan repository root, add the checkout to `PYTHONPATH`. Monarch-spawned RL worker processes inherit this environment variable, so they can import the local `torchtitan` package:
 ```bash
-uv pip install -e .
+export PYTHONPATH="$PWD:${PYTHONPATH:-}"
```
Comment (felipemello1) [inline torchtitan/experiments/rl/README.md:65]: yeah, i never did this, and it works fine. Perhaps this is necessary for `worktree`? I think that for worktree this is actually necessary.

can you ask claude to create a new env and verify this claim (i.e. do with it and without it, confirm that it fails without it?)

but in any case, we would still need to run uv pip install -e no?

File: .github/workflows/integration_test_4gpu_rl.yaml [hunk 4]
```diff
@@ -79,9 +79,8 @@ jobs:
           --extra-index-url https://download.pytorch.org/whl/nightly/cu130 \
           --index-strategy unsafe-best-match
 
-        # 4. Install torchtitan in editable mode because integration test spawns
-        # python grpo.py as a subprocess
-        uv pip install -e .
+        # 4. Make the checkout importable for subprocesses spawned by the test.
+        export PYTHONPATH="$PWD:${PYTHONPATH:-}"
```
Comment (felipemello1) [inline .github/workflows/integration_test_4gpu_rl.yaml:84]: same here, wouldnt we need to still run pip install -e? otherwise, how is torchtitan installed?

Comment (tianyu-l) [reply .github/workflows/integration_test_4gpu_rl.yaml:84]: torchtitan doesn't need to be installed to run?

Comment (felipemello1) [reply .github/workflows/integration_test_4gpu_rl.yaml:84]: i have always used `pip install -e .` as a way for programs to find imports, right?  as if the library is installed and is importable. "from torchtitan.X import Y"

I guess the export PYTHONPATH also works? i just haven't seen similar instruction in other libraries. So i dont know if this is a good solution, or claude slop

But if both of you are confident, lets go with it

Comment (wwwjn) [reply .github/workflows/integration_test_4gpu_rl.yaml:84]: >  wouldnt we need to still run pip install -e? 

No, export PYTHONPATH also works for the same purpose of "from trochtitan.X import Y",  We don't use `pip install -e . ` because it will have permission issue in CI, eg: https://github.com/pytorch/torchtitan/actions/runs/27032930788/job/79789585434#step:17:14902 This stops us running the CI.

Comment (wwwjn) [reply .github/workflows/integration_test_4gpu_rl.yaml:84]: > torchtitan doesn't need to be installed to run?

The main purpose we'll have this is for Monarch / vllm spawned children process - The children process need to know where to `import torchtitan`. To let them know the torchtitan path, we'd put it in a env variable which will be inherited by child process, or install torchtitan via editable mode `pip install -e .`

I don't know if there's a better work around, but the env variable could be a possible one

**Review by felipemello1 — APPROVED** [review]

**Review by tianyu-l — COMMENTED** [review]
unblock

**Review by tianyu-l — APPROVED** [review]

File: torchtitan/experiments/rl/train.py [hunk 5]
```diff
@@ -17,8 +17,8 @@
    in the controller.
 
 Command to run:
-python3 torchtitan/experiments/rl/train.py \
-    --module rl --config rl_grpo_qwen3_0_6b \
```
Comment (wwwjn) [inline torchtitan/experiments/rl/train.py:21]: This config field is renamed to rl_grpo_qwen3_0_6b_varlen, but test is not updated

File: torchtitan/experiments/rl/tests/integration_tests.py [hunk 6]
```diff
@@ -33,11 +33,13 @@ def build_rl_test_list() -> list[OverrideDefinitions]:
...
-                    "--config rl_grpo_qwen3_0_6b",
+                    "--config rl_grpo_qwen3_0_6b_varlen",
                     "--trainer.parallelism.tensor_parallel_degree 2",
                     "--generator.parallelism.tensor_parallel_degree 2",
-                    "--generator.sampling.n 2",
-                    "--batcher.batch.seq_len 256",
+                    "--group_size 2",
+                    "--batcher.batch.seq_len 1024",
+                    "--renderer.enable-thinking False",
+                    "--generator.sampling.max_tokens 256",
```
Comment (wwwjn) [inline torchtitan/experiments/rl/tests/integration_tests.py:42]: Increase the number of tokens being generated, to include meaningful answer to the SumDigts task

File: torchtitan/experiments/rl/actors/trainer.py [hunk 7]
```diff
@@ -264,9 +269,14 @@ def load_state_dict(self, state_dict: dict[str, Any]) -> None:
 
     @endpoint
     async def close(self) -> None:
-        """Destroy the worker's torch.distributed process group."""
-        if torch.distributed.is_initialized():
-            torch.distributed.destroy_process_group()
+        """Close actor-local resources before the process mesh stops.
```
Comment (wwwjn) [inline torchtitan/experiments/rl/actors/trainer.py:272]: This change is because CI hang during tear down, so I vibe coded this solution which looks reasonable - cc @felipemello1 @pzhan9 , I might want to land this PR first and if it breaks the shutdown, would love to have a forward fix
